--- scripts/package-cli.orig	2020-11-21 16:53:26.463303591 -0800
+++ scripts/package-cli	2020-11-21 16:54:32.435724124 -0800
@@ -40,16 +40,11 @@
 
 cp ./build/out/data.tar.gz ./build/data/${HASH}.tgz
 
-BIN_SUFFIX="-${ARCH}"
-if [ ${ARCH} = amd64 ]; then
-    BIN_SUFFIX=""
-elif [ ${ARCH} = arm ]; then
-    BIN_SUFFIX="-armhf"
-fi
+BIN_SUFFIX=""
 
 CMD_NAME=dist/artifacts/k3s${BIN_SUFFIX}
 
-"${GO}" generate
+CGO_ENABLED=0 GOARCH= "${GO}" generate
 LDFLAGS="
     -X github.com/rancher/k3s/pkg/version.Version=$VERSION
     -X github.com/rancher/k3s/pkg/version.GitCommit=${COMMIT:0:8}
